WalletWasabi/Blockchain/Transactions/TransactionStore.cs

```diff
@@ -1,411 +1,258 @@
 using NBitcoin;
-using Nito.AsyncEx;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using WalletWasabi.Blockchain.Transactions.Operations;
 using WalletWasabi.Extensions;
 using WalletWasabi.Helpers;
-using WalletWasabi.Io;
 using WalletWasabi.Logging;
-using WalletWasabi.Nito.AsyncEx;
+using WalletWasabi.Stores;
 
 namespace WalletWasabi.Blockchain.Transactions;
 
 public class TransactionStore : IAsyncDisposable
 {
-	public TransactionStore(string workFolderPath, Network network)
+	public TransactionStore(string workFolderPath, Network network, bool migrateData = true)
 	{
-		WorkFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
-		Network = network;
+		workFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
 
-		// In Transactions.dat every line starts with the tx id, so the first character is the best for digest creation.
-		TransactionsFileManager = new IoManager(filePath: Path.Combine(WorkFolderPath, "Transactions.dat"));
+		if (workFolderPath == SqliteStorageHelper.InMemoryDatabase)
```

Comment:
The code is cleaner if you add that this is only used in the tests. Otherwise, ppl like me think is has something to do with the prod and start the investigation.